ansible/playbooks/adhoc/hotfix/roles/update_master/tasks/main.yml

```diff
@@ -0,0 +1,70 @@
+---
+
+# Default to False before attempting upgrade
+- set_fact: 
+    update_successful:    False
+    downgrade_successful: False
+  when: "{{ update_needed == True }}" 
+
+- name: Update atomic-openshift-master
+  yum:
+    name: "{{ rpm_wanted }}"
+    state: present
+  when: "{{ update_needed == True }}" 
+
+- name: Check if update is successful
+  command: "rpm -qa atomic-openshift"
+  register: post_upgrade
+  changed_when: False
+
+# ansible will return a '17' if a match is found. '-1' means 'not found'
+- name: Set update_successful=True if upgrade succeeded
+  set_fact: 
+    update_successful: True
+  when: "{{ post_upgrade.stdout.find(version_wanted) != -1 }}"
+
+- name: Upgrade failed, so try a downgrade
+  command: "yum -y downgrade atomic-openshift-{{ version_wanted }} atomic-openshift-clients-{{ version_wanted }} atomic-openshift-master-{{ version_wanted }} atomic-openshift-node-{{ version_wanted }} tuned-profiles-atomic-openshift-node-{{ version_wanted }} atomic-openshift-sdn-ovs-{{ version_wanted }}"
+  when: "{{ update_successful == False }}"
```

Comment:
@dak1n1 ok, I like your logic on this. I researched it a little, and apparently the yum ansible module before ansible 1.9 allowed you do downgrade. But the current one doesn't allow it.

So, given that, what's the roll back plan then?

Is it re-run this, but with a different version (presumably with the current version installed)?